File: 06__full-stack/academy-os/backend/internal/academies/academy.go
package academies

import (
	"context"
)

type Academy struct {
	ID           string  `json:"id"`
	OwnerUserID  string  `json:"owner_user_id"`
	Name         string  `json:"name"`
	Slug         string  `json:"slug"`
	Subdomain    string  `json:"subdomain"`
	CustomDomain *string `json:"custom_domain"`
	Status       string  `json:"status"`
}

type academyCtxKey struct{}

var contextKey = academyCtxKey{}

func WithContext(ctx context.Context, academy *Academy) context.Context {
	return context.WithValue(ctx, contextKey, academy)
}

func FromContext(ctx context.Context) (*Academy, bool) {
	academy, ok := ctx.Value(contextKey).(*Academy)
	return academy, ok
}

File: 06__full-stack/academy-os/frontend/app/types/academy.ts
export interface Academy {
  id: string
  ownerUserId?: string
  name: string
  slug: string
  subdomain?: string
  customDomain?: string
  status?: string
  description?: string
  logoUrl?: string
  faviconUrl?: string
  primaryColor?: string
  secondaryColor?: string
  createdAt?: string
  updatedAt?: string
}

export type Tenant = Academy
